module/zfs/dsl_dir.c

```diff
@@ -2124,6 +2124,10 @@ dsl_dir_rename_sync(void *arg, dmu_tx_t *tx)
 	VERIFY0(zap_add(mos, dsl_dir_phys(newparent)->dd_child_dir_zapobj,
 	    dd->dd_myname, 8, 1, &dd->dd_object, tx));
 
+#ifdef _KERNEL
+	zfsvfs_update_fromname(ddra->ddra_oldname, ddra->ddra_newname);
+#endif
```

Comment:
The layering violation here isn't great, but I suppose it's really no worse that the call to `zvol_rename_minors()` below.  Let me just suggest that you add an empty `zfsvfs_update_fromname()` function to `lib/libzpool/kernel.c` so at least the `#ifdef _KERNEL` can be removed.